mixer/pkg/protobuf/yaml/encoder.go

```diff
@@ -63,69 +63,909 @@ func (e *Encoder) EncodeBytes(data map[interface{}]interface{}, msgName string,
+		if packed {
+			/*
+				dAtA[i] = 0xba
+				i++
+				dAtA[i] = 0x1
+				i++
+				i = encodeVarintTypes(dAtA, i, uint64(len(m.RDbl)*8))
+				for _, num := range m.RDbl {
+					f4 := math.Float64bits(float64(num))
+					encoding_binary.LittleEndian.PutUint64(dAtA[i:], uint64(f4))
+					i += 8
+				}
+			*/
+			v, ok := data.([]interface{})
+			if !ok {
+				return badTypeError(name, "[]float64", data)
+			}
+			_ = buffer.EncodeVarint(encodeIndexAndType(fieldNumber, wireType))
+			_ = buffer.EncodeVarint(uint64(len(v) * 8))
+			for i, iface := range v {
+				c, ok := toFloat(iface)
+				if !ok {
+					return badTypeError(fmt.Sprintf("%s[%d]", name, i), "float64", iface)
+				}
+				_ = buffer.EncodeFixed64(math.Float64bits(c))
+			}
+		} else if repeated {
+			/*
+				for _, num := range m.RDblUnpacked {
+					dAtA[i] = 0xc1
+					i++
+					dAtA[i] = 0x1
+					i++
+					f5 := math.Float64bits(float64(num))
+					encoding_binary.LittleEndian.PutUint64(dAtA[i:], uint64(f5))
+					i += 8
+				}
+			*/
+			v, ok := data.([]interface{})
+			if !ok {
+				return badTypeError(name, "[]float64", data)
+			}
+
+			for i, iface := range v {
+				c, ok := toFloat(iface)
+				if !ok {
+					return badTypeError(fmt.Sprintf("%s[%d]", name, i), "float64", iface)
+				}
+				_ = buffer.EncodeVarint(encodeIndexAndType(fieldNumber, wireType))
+				_ = buffer.EncodeFixed64(math.Float64bits(c))
+			}
+		} else {
+			/*
+				dAtA[i] = 0x11
+				i++
+				encoding_binary.LittleEndian.PutUint64(dAtA[i:], uint64(math.Float64bits(float64(m.Dbl))))
+				i += 8
+			*/
+			v, ok := toFloat(data)
+			if !ok {
+				return badTypeError(name, "float64", data)
+			}
+			_ = buffer.EncodeVarint(encodeIndexAndType(fieldNumber, wireType))
+			_ = buffer.EncodeFixed64(math.Float64bits(v))
 		}
 
-		// Errors from proto.Buffer.Encode* are always nil, therefore ignoring.
-		_ = buffer.EncodeVarint(encodeIndexAndType(int(field.GetNumber()), proto.WireBytes))
-		_ = buffer.EncodeStringBytes(v)
-	case descriptor.FieldDescriptorProto_TYPE_DOUBLE:
-		v, ok := data.(float64)
-		if !ok {
-			return badTypeError(name, "float64", data)
+	case descriptor.FieldDescriptorProto_TYPE_FLOAT:
+		if packed {
+			/*
+				dAtA[i] = 0xf2
+				i++
+				dAtA[i] = 0x1
+				i++
+				i = encodeVarintTypes(dAtA, i, uint64(len(m.RFlt)*4))
+				for _, num := range m.RFlt {
+					f8 := math.Float32bits(float32(num))
+					encoding_binary.LittleEndian.PutUint32(dAtA[i:], uint32(f8))
+					i += 4
+				}
+			*/
+			v, ok := data.([]interface{})
+			if !ok {
+				return badTypeError(name, "[]float32", data)
+			}
+			_ = buffer.EncodeVarint(encodeIndexAndType(fieldNumber, wireType))
+			_ = buffer.EncodeVarint(uint64(len(v) * 4))
+			for i, iface := range v {
+				c, ok := toFloat(iface)
+				if !ok {
+					return badTypeError(fmt.Sprintf("%s[%d]", name, i), "float32", iface)
+				}
+				_ = buffer.EncodeFixed32(uint64(math.Float32bits(float32(c))))
+			}
+		} else if repeated {
```

Comment:
Minor nit: packed fields are always repeated right? So, logically, it looks like a structure of "if repeated { if packed {} else {} }" follows the logical model better.

Reply:
I have tried to follow the code flow in gogo marshal code, so that I don't miss out on any case. Therefore, I have this structuring. This is useful for me to go back to their code and see what they are doing in codegen and fix any issues here.